### PR TITLE
connect with minecraft account

### LINKED ISSUE #283: Connection with minecraft account
To advance some current and make some future features possible the backend needs to know what minecraft account corresponds to a user. 
The `conMc(uuid)` was created for this. It returns a number `1-999`. 
The user should put up or bid on an auction with the last 3 digits set to that number to verify that he owns that minecraft account.
The option to connect a minecraft account is best suited for `/player/{uuid}`

### PR DESCRIPTION
**Pull request by matthias-luger**
fixes #283

### CODE REVIEW

**Comment by Ekwav** [comment]
What kind of help is wanted by you @matthias-luger ?

**Comment by Flou21** [comment]
[CI] created a pr environment: https://pr-env-sky-317.flou21.de

**Comment by Flou21** [comment]
[CI] created a pr environment: https://pr-env-sky-317.flou21.de

**Comment by Flou21** [comment]
[CI] created a pr environment: https://pr-env-sky-317.flou21.de

**Comment by Flou21** [comment]
[CI] created a pr environment: https://pr-env-sky-317.flou21.de

**Comment by matthias-luger** [comment]
@Ekwav The accountInfo command returns a error. I dont know what I am doing wrong. Other than that I only need the text what the feature currently does for the user. Then its ready to merge.

**Comment by Flou21** [comment]
[CI] created a pr environment: https://pr-env-sky-317.flou21.de

**Comment by Ekwav** [comment]
> @Ekwav The accountInfo command returns a error. I dont know what I am doing wrong. Other than that I only need the text what the feature currently does for the user. Then its ready to merge.

The account info was missing a default constructor thus the serialization threw an exception. I fixed that.
Currently I think the following would be a good description:
> Connecting a your Minecraft account allows us to improve your experience. There are a lot of features in development where we need to know what your Minecraft account is. Currently connecting your account only adds your Minecraft name to the link preview when you share your referral link, we don't store your Name and publicly displaying your email isn't a good choice either.

**Comment by Flou21** [comment]
[CI] created a pr environment: https://pr-env-sky-317.flou21.de

File: src/pages/PlayerDetails/ClaimAccount/ClaimAccount.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            {isLoggedIn ?
+                <div>
+                    <h3>How do I claim my Minecraft Account?</h3>
+                    <p>Put a bid or an auction ending with the number shown below in the auction house <b>within 10 minutes</b>. After a short period we will be able to verify that this account belongs to you.</p>
+                    <h4>Your number: <b style={{ color: "lime" }}>{verificationNumber}</b></h4>
+                    <hr />
+                    <h3>Example</h3>
+                    <p>You can verify your account by these actions (examples):</p>
+                    <ul>
+                        <li>Bid <b>{verificationNumber}</b> coins on an auction</li>
```
Comment (Ekwav) [inline src/pages/PlayerDetails/ClaimAccount/ClaimAccount.tsx:40]: The verificationNumber should be left padded with 0 
image

File: src/pages/PlayerDetails/ClaimAccount/ClaimAccount.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            setIsLoading(false);
+        });
+    }
+
+    return (
+        <div className="claim-account">
+            {isLoggedIn ?
+                <div>
+                    <h3>How do I claim my Minecraft Account?</h3>
+                    <p>Put a bid or an auction ending with the number shown below in the auction house <b>within 10 minutes</b>. After a short period we will be able to verify that this account belongs to you.</p>
```
Comment (Ekwav) [inline src/pages/PlayerDetails/ClaimAccount/ClaimAccount.tsx:34]:
```suggestion
                    <p>In order to verify that his account is yours you have to create a bid or an auction with the last 3 digits set to the number below.  After at most 3 minutes your account will automatically be verified by our server. If your account is not verified withing 5 minutes please try again.</p>
```

File: src/pages/PlayerDetails/ClaimAccount/ClaimAccount.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+                        <li>Create an auction for <b>{verificationNumber}</b> coins</li>
+                        <li>Create an auction for 53.<b>{verificationNumber}</b> coins</li>
+                        <li>. . .</li>
+                    </ul>
+                    <hr />
+                    <h3>Why?</h3>
+                    <p>Connecting your Minecraft account allows us to improve your experience.
+                        There are a lot of features in development where we need to know what your Minecraft account is.
+                        Currently connecting your account only adds your Minecraft name to the link preview when you share your <Link to="/ref">referral link</Link>.
+                        We don't store your name and publicly displaying your email isn't a good choice either.</p>
```
Comment (Ekwav) [inline src/pages/PlayerDetails/ClaimAccount/ClaimAccount.tsx:51]: I think we should give our users the information what to expect.
```suggestion
                        Future features include tracking your profit from the flipper and hiding your history.</p>
```

File: src/pages/PlayerDetails/ClaimAccount/ClaimAccount.tsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+                </div> : isLoading ? getLoadingElement() :
+                    <div>
+                        <p>To claim your Minecraft Account please log in with Google:</p>
+                    </div>
+            }
+            <GoogleSignIn onAfterLogin={onAfterLogin} />
```
Comment (Ekwav) [inline src/pages/PlayerDetails/ClaimAccount/ClaimAccount.tsx:57]: is this just here to get the callback? This button is never displayed.

File: src/pages/PlayerDetails/PlayerDetails.tsx [hunk 5]
```diff
@@ -61,10 +67,53 @@ function PlayerDetails() {
...
 
+    function onAfterLogin() {
+        api.getAccountInfo().then(info => {
+            setAccountInfo(info);
+        })
+    }
+
+    let claimAccountElement = (
+        !wasAlreadyLoggedIn() ? null :
+            selectedPlayer?.name !== accountInfo?.mcName ?
```
Comment (Ekwav) [inline src/pages/PlayerDetails/PlayerDetails.tsx:78]: Is it possible to compare based on the uuid? 
the name may be changed and could result in timing issues

File: src/pages/PlayerDetails/ClaimAccount/ClaimAccount.tsx [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+                        <li>Create an auction for 53.<b>{verificationNumber}</b> coins</li>
+                        <li>. . .</li>
+                    </ul>
+                    <hr />
+                    <h3>Why?</h3>
+                    <p>Connecting your Minecraft account allows us to improve your experience.
+                        There are a lot of features in development where we need to know what your Minecraft account is.
+                        Currently connecting your account only adds your Minecraft name to the link preview when you share your <Link to="/ref">referral link</Link>.
+                        We don't store your name and publicly displaying your email isn't a good choice either.</p>
+                </div> : isLoading ? getLoadingElement() :
```
Comment (Ekwav) [inline src/pages/PlayerDetails/ClaimAccount/ClaimAccount.tsx:52]: There should either be a button "I created the bid. please verify me" or an auto request to `accountInfo` (every ~30 sec) to see if the verification was completed. 
Currently there is no indication when an user gets verified

File: src/api/ApiHelper.tsx [hunk 7]
```diff
@@ -758,13 +759,57 @@ function initAPI(): API {
                     resolve(data.map(a => parseFilterOption(a)));
                 },
                 reject: function (error) {
-                    apiErrorHandler(RequestType.ACTIVE_AUCTIONS, error, item.tag);
+                    apiErrorHandler(RequestType.FILTER_FOR, error, item.tag);
```
Comment (Ekwav) [inline src/api/ApiHelper.tsx:762]: merge conflict

**Review by Ekwav — CHANGES_REQUESTED** [review]

File: src/pages/PlayerDetails/ClaimAccount/ClaimAccount.tsx (hunk 4 above)
Comment (matthias-luger) [reply src/pages/PlayerDetails/ClaimAccount/ClaimAccount.tsx:57]: This component handles the entire login logic, it doesnt need to show the button, but should login the user if he ever was logged in

**Comment by matthias-luger** [comment]
Has to be looked at, after the backend extended the mcInfo-Response

**Comment by Ekwav** [comment]
> Has to be looked at, after the backend extended the mcInfo-Response

done

**Comment by Ekwav** [comment]
This should target the branch `develop`

**Pull request merged**